cleaner/cleaner.go

```diff
@@ -0,0 +1,38 @@
+package cleaner
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+type t func() error
+
+var cleaners []t
+var inactive bool
+
+// Register a cleaner function. When a function is registered, the Signal watcher is started in a goroutine.
+func Register(f t) {
+	cleaners = append(cleaners, f)
+	if !inactive {
+		inactive = false
+		go func() {
+			ch := make(chan os.Signal, 1)
+			signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
+			<-ch
+			fmt.Println("")
+			clean()
+			os.Exit(1)
+		}()
+	}
+}
+
+// Clean invokes all registered cleanup functions
+func clean() {
+	fmt.Println("Cleaning")
+	for _, f := range cleaners {
+		_ = f()
```

Comment:
Print the error if any ?

Reply:
I believe printing errors here would be unnecessary.

First, if the user is shutting the command down by force, it is likely that they do not want to see a bunch of information on the terminal.
Second but more important : If previous lockfiles had alredy been removed from disk by the deferables, a "file now found" error will occour, but that does not affect the user in any way. Implementing a "removal" function to remove previously registered cleaners that have became unnecessary is going to add complexity, and I see no real benefit.

Tell me what you prefere: leve it as is, print errors as is, or add the removal functionality.

Reply:
> You are writing a reusable package.

To make it more reusable, I am considering separating the anonymous function and giving it a name. 
It would be possible to summon it manually, or parsing it to the RegisterCleaner as an argument. 
```
func RegisterCleaner(watcher func(), f ...Cleaner) {
	for _, fn := range f {
		cleaners = append([]Cleaner{fn}, cleaners...)
		if !active {
			active = true
			go watcher()
		}
	}
}
```

The calling packages would do something like : 
```
interrupt.RegisterCleaner( interrupt.SignalWatcher, backend.Close)
```

It would be easy to create different watchers to deal with the errors in a different way, print in a different way...

How do you like this approach ?

Reply:
It's highly subjective, but I'd say that it would be over engineering. There is no need for that currently in the project and I don't foresee something that would use it, so better keep things simple.